### ISSUE TITLE
Remove the User CRD

### ISSUE THREAD

**Issue by sebgl** [elastic/cloud-on-k8s#727]
It's an extra "internal" CRD users might wonder about.
We don't plan to support users creating their own users through the user CRD.

Internally, we mostly use it for the Kibana user. This can be replaced by a user name convention + secret holding password details. Similar to what we do for other internal users.

**Comment by barkbay** [comment]
Is there something else to discuss on this one or someone can work it immediately ?

**Comment by sebgl** [comment]
Nothing else to discuss imo, go for it! 🚀

**Comment by barkbay** [comment]
Just to share the idea I'm working on: the user CRD would be replaced by a secret that lives in the Kibana namespace.
The user and the password are stored as a key/value pair in the secret, it is consistent with what is done today for other users and compatible with the way the secret is referenced in the Kibana crd.
The name and the namespace of the ES cluster it belongs to are stored as some annotations.

Here is an example:

```yaml
apiVersion: v1
data:
  ns2-kibana-sample-kibana-user: alongandrandompassword
kind: Secret
metadata:
  annotations:
    user.k8s.elastic.co/roles: kibana_system
  labels:
    common.k8s.elastic.co/type: user
    elasticsearch.k8s.elastic.co/cluster-name: elasticsearch-sample
    elasticsearch.k8s.elastic.co/cluster-namespace: default
    kibana.k8s.elastic.co/name: kibana-sample
    kibanaassociation.k8s.elastic.co/name: kibana-sample
  name: kibana-sample-kibana-user
  namespace: ns2
  ownerReferences:
  - apiVersion: kibana.k8s.elastic.co/v1alpha1
    blockOwnerDeletion: true
    controller: true`
    kind: Kibana
    name: kibana-sample
type: Opaque
```

**Comment by sebgl** [comment]
@barkbay do we need to "duplicate" this into ES namespace for the ES controller to inject it into ES configuration?

**Comment by barkbay** [comment]
I don't think so _(at least that's not the idea)_, it's the reason I add the `elasticsearch.k8s.elastic.co/cluster-namespace` and `elasticsearch.k8s.elastic.co/cluster-name` annotations, the ES controller should be able to watch the secret and figure out which ES cluster must update its configuration.

**Comment by pebrc** [comment]
I am not sure this would work there is no guarantee that the elasticsearch controller in the Kibana namespace has access to the elasticsearch cluster which might be in another namespace. Vice versa the elasticsearch controller in the namespace where elasticsearch resides might not be able to read the secret in the Kibana namespace

**Comment by barkbay** [comment]
fixed by #948
